File: 21560 - Sai Suraj/week 8/34_Find First and Last Position of Element in Sorted Array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        left = self.BS(nums,target,True)
        right = self.BS(nums,target,False)
        return [left,right]
    def BS(self,nums,target,leftbias):
        l = 0
        r = len(nums)-1
        i=-1
        while(l<=r):
            m =(l+r)//2
            if target > nums[m]:
                l=m+1
            elif target<nums[m]:
                r=m-1
            else:
                i=m
                if leftbias:
                    r=m-1
                else:
                    l=m+1
        return i

File: 21558 - Nithin/week 8/34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int len = nums.size();
        int left = 0, right = len -1;
        int mid, res = -1;
        vector<int> pos;
        while(left<=right){
            mid = left + (right -left)/2;
            if(nums[mid] == target){
                res = mid;
                right = mid-1;
                
            }
            
            else if(nums[mid] < target)
                left = mid+1;
            else
                right = mid-1;
            
        }
        pos.push_back(res);
        
        left = 0;
        right = len -1;
        res = -1;
        
        while(left<=right){
            mid = left + (right -left)/2;
            if(nums[mid] == target){
                res = mid;
                left = mid+1;
                
            }
            
            else if(nums[mid] < target)
                left = mid+1;
            else
                right = mid-1;
            
        }
        pos.push_back(res);
        return pos;
    }
};